boards.txt

```diff
@@ -35828,4 +35828,155 @@ Geekble_ESP32C3.menu.EraseFlash.none.upload.erase_cmd=
 Geekble_ESP32C3.menu.EraseFlash.all=Enabled
 Geekble_ESP32C3.menu.EraseFlash.all.upload.erase_cmd=-e
 
-##############################################################
\ No newline at end of file
+##############################################################
+
+waveshare_esp32s3_touch_lcd_128.name=Waveshare ESP32S3 Touch LCD 128
+waveshare_esp32s3_touch_lcd_128.vid.0=0x1a86
+waveshare_esp32s3_touch_lcd_128.pid.0=0x55d3
+
+waveshare_esp32s3_touch_lcd_128.upload.tool=esptool_py
+waveshare_esp32s3_touch_lcd_128.upload.tool.default=esptool_py
+waveshare_esp32s3_touch_lcd_128.upload.tool.network=esp_ota
+waveshare_esp32s3_touch_lcd_128.upload.maximum_size=16777216
+waveshare_esp32s3_touch_lcd_128.upload.maximum_data_size=327680
+waveshare_esp32s3_touch_lcd_128.upload.wait_for_upload_port=true
+waveshare_esp32s3_touch_lcd_128.upload.speed=460800
+waveshare_esp32s3_touch_lcd_128.upload.flags=
+waveshare_esp32s3_touch_lcd_128.upload.extra_flags=
+
+waveshare_esp32s3_touch_lcd_128.bootloader.tool=esptool_py
+waveshare_esp32s3_touch_lcd_128.bootloader.tool.default=esptool_py
+
+waveshare_esp32s3_touch_lcd_128.serial.disableDTR=true
+waveshare_esp32s3_touch_lcd_128.serial.disableRTS=true
+
+waveshare_esp32s3_touch_lcd_128.build.tarch=xtensa
+waveshare_esp32s3_touch_lcd_128.build.bootloader_addr=0x0
+waveshare_esp32s3_touch_lcd_128.build.mcu=esp32s3
+waveshare_esp32s3_touch_lcd_128.build.core=esp32
+waveshare_esp32s3_touch_lcd_128.build.target=esp32s3
+waveshare_esp32s3_touch_lcd_128.build.variant=waveshare_esp32s3_touch_lcd_128
+waveshare_esp32s3_touch_lcd_128.build.board=WAVESHARE_ESP32S3_TOUCH_LCD_128
+
```

Comment:
```suggestion

waveshare_esp32s3_touch_lcd_128.build.usb_mode=1
waveshare_esp32s3_touch_lcd_128.build.cdc_on_boot=1
waveshare_esp32s3_touch_lcd_128.build.msc_on_boot=0
waveshare_esp32s3_touch_lcd_128.build.dfu_on_boot=0
```

Reply:
Is this need if the board uses an external USB to UART chip?

Reply:
Those are functions of USB are related to the USB-OTG mode, which can be used. So yes :)

Reply:
So with the default settings we added:
```
waveshare_esp32s3_touch_lcd_128.build.usb_mode=1
waveshare_esp32s3_touch_lcd_128.build.cdc_on_boot=1
waveshare_esp32s3_touch_lcd_128.build.msc_on_boot=0
waveshare_esp32s3_touch_lcd_128.build.dfu_on_boot=0
```
The board is working fine right? You can see the output in the Serial monitor.

Reply:
I can see serial output if I use Serial1. The built-in USB connection is not broken out.

Reply:
board is serial-only. Uses a crappy WCH chip for USB to Serial.

Reply:
Yep.